### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by olaishola05**
# Decorate Class

- Create an interface.rb file 
- Create a class Nameable.
   - Implement a method called correct_name that raises a NotImplementedError.
- Refactor Person class to inherits from Nameable and has a method correct_name implemented that returns the name attributes
- Create a BaseDecorator class that inherits from Nameable with a constructor with params @nameable and method  correct_name which returns @nameable
- Create CapitalizeDecorator and TrimmerDecorator class that inherits from Nameable, a method correct_name one returns a capitalized name and the other checks & trim if name passed if longer than 10 characters

### CODE REVIEW

File: interface.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class BaseDecorator
```
Comment (Mihndim2020) [inline interface.rb]: - [ ] The `BaseDecorator class` inherits from the `Nameable class`, kindly fix that. Please make sure you also set the getter and setter methods for the nameable object. Something like so: 
 `class BaseDecorator < Nameable` , `attr_accessor :nameable`.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages,
:heavy_check_mark: Great job with Gitflow.

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

### I will leave these comments here, since the concern your readme file on which you have not made any changes for this particular `PR`

- [ ] Please kindly update your readme file to reflect the changes for this milestone, as this will make your readme file more informative and professional. Please checkout this [template](https://github.com/microverseinc/readme-template) and adapt it to your project.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: interface.rb (hunk 1 above)
Comment (olaishola05) [reply interface.rb]: Thank you @Mihndim2020 for the tips.

**Review by t-yanick — APPROVED** [review]
Hi @netman5, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by olaishola05** [comment]
Thank you @t-yanick

**Pull request merged**